<a href="https://colab.research.google.com/github/gabrielreis35/Lab2-pds/blob/master/Lab2_PDS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import matplotlib.pyplot as plt
import soundfile as sf
from scipy.fftpack import fft

In [ ]:
file_path = '/content/Nota 02.wav'
audio_signal, sample_rate = sf.read(file_path)

In [ ]:
len(audio_signal.shape)

2

In [ ]:
if len(audio_signal.shape) > 1:
    audio_signal = audio_signal[:, 0]

In [ ]:
n = len(audio_signal)
audio_fft = fft(audio_signal)
frequencies = np.fft.fftfreq(n, 1/sample_rate)

In [ ]:
magnitude = np.abs(audio_fft)
frequency_index = np.argmax(magnitude)
dominant_frequency = abs(frequencies[frequency_index])

In [ ]:
frequencies[frequency_index]

418.04639660857276

# A nota 1 (frequencia 443) é referente à nota A e a nota 2 (frequencia 418) é referente à nota B